### DISCUSSION TITLE
rpmbuild -ta option no longer works

### DISCUSSION THREAD

**Discussion by aditigaur4** [rpm-software-management/rpm#3453] [Q&A]
`rpmbuild -ta <tarball>` complains that it fails to find the spec file. The spec file exists in the top of the unpacked directory.

````
root [ / ]# rpmbuild -ta slurm-24.05.4.tar.bz2
error: Failed to read spec file from slurm-24.05.4.tar.bz2
````
The slurm tarball in this case does have the spec file at the top:

````
root [ / ]# tar -xvf slurm-24.05.4.tar.bz2 | grep slurm.spec
slurm-24.05.4/slurm.spec
root [ / ]#
````

This is rpmbuild version 4.17 but the problem exists in 4.18 as well. Note this is mariner OS. 

````
root [ / ]# cat /etc/os-release
NAME="Common Base Linux Mariner"
VERSION="2.0.20241029"
ID=mariner
VERSION_ID="2.0"
PRETTY_NAME="CBL-Mariner/Linux"
ANSI_COLOR="1;34"
HOME_URL="https://aka.ms/cbl-mariner"
BUG_REPORT_URL="https://aka.ms/cbl-mariner"
SUPPORT_URL="https://aka.ms/cbl-mariner"
````

This same command works in rpmbuild version 4.16:

````
[root@341759f80574 /]# rpmbuild -ta slurm-24.05.4.tar.bz2
warning: Macro expanded in comment on line 22: %_prefix path            install path for commands, libraries, etc.

warning: Macro expanded in comment on line 209: %define _unpackaged_files_terminate_build      0
````

````
[root@341759f80574 /]# rpmbuild --version
RPM version 4.16.1.3
````

I did not notice any changes to this specific flag in the release notes, any help on this would be greatly appreciated, 
Thank you!

**Comment by pmatilai** [comment]
Can't reproduce that here, this is almost certainly some local/distro issue. My guess is that bzip2 either missing on the system or the macro for it is pointing to a wrong thing, and in both cases that's something that should be reported on Mariner, we can't help that (and 4.17-4.18 are out of upstream support by now anyway, so converting this to a discussion).

`rpm --eval "%{uncompress:slurm-24.05.4.tar.bz2}"`  should output something like `/usr/bin/bzip2 -dc slurm-24.05.4.tar.bz2` and executing the output'ed command should work.
